Bus Scheduler/src/store/api/axios.ts

```diff
@@ -0,0 +1,59 @@
+import axios, { AxiosError } from "axios";
+import { baseApiUrl } from "./urls";
+
+
+const baseApi = axios.create({
+  baseURL: baseApiUrl,
+  headers: {
+    'Content-Type': 'application/json',
+  },
+});
+
+
+export const GET = <T>(url: string): Promise<T> => {
+
+  return baseApi.get(url)
+    .then(response => {
+      const responseData: T = response?.data?.data
+      return responseData
+    })
+
+}
+
+// Request interceptor
+baseApi.interceptors.request.use(
+  (config) => {
+    return config;
+  },
+  (error) => {
+    return Promise.reject(error);
+  },
+);
+
+
+// Response interceptor
+baseApi.interceptors.response.use(
```

Comment:
good to use seperate files for interceptors

Reply:
Done